packages/@ama-sdk/schematics/schematics/typescript/core/index.ts

```diff
@@ -36,119 +36,175 @@ const getPathObjectTemplate = (pathObj: PathObject) => {
+        }
+      }
 
-  /**
-   * Update local swagger spec file
-   * @param tree
-   * @param _context
-   */
-  const updateSpec = (tree: Tree, _context: SchematicContext) => {
-    const readmeFile = path.posix.join(targetPath, 'readme.md');
-    const specContent = readFileSync(specPath).toString();
-    if (tree.exists(readmeFile)) {
-      const swaggerVersion = /version: ([0-9]+\.[0-9]+\.[0-9]+)/.exec(specContent);
-
-      if (swaggerVersion) {
-        const readmeContent = tree.read(readmeFile)!.toString('utf8');
-        tree.overwrite(readmeFile, readmeContent.replace(/Based on Swagger spec .*/i, `Based on Swagger spec ${swaggerVersion[1]}`));
+      if (tree.exists(path.posix.join(targetPath, 'swagger-spec.yaml'))) {
+        tree.overwrite(path.posix.join(targetPath, 'swagger-spec.yaml'), specContent);
+      } else {
+        tree.create(path.posix.join(targetPath, 'swagger-spec.yaml'), specContent);
       }
-    }
+      return () => tree;
+    };
 
-    if (tree.exists(path.posix.join(targetPath, 'swagger-spec.yaml'))) {
-      tree.overwrite(path.posix.join(targetPath, 'swagger-spec.yaml'), specContent);
-    } else {
-      tree.create(path.posix.join(targetPath, 'swagger-spec.yaml'), specContent);
-    }
-    return () => tree;
-  };
+    const JAVA_OPTIONS = ['specPath', 'specConfigPath', 'globalProperty'];
 
-  const runGeneratorRule = (tree: Tree, context: SchematicContext) => {
-    const generatorOptions: Partial<OpenApiCliOptions> = {specPath, globalProperty};
-    const packageJsonFile: {openApiSupportedVersion?: string} = JSON.parse((readFileSync(path.join(__dirname, '..', '..', '..', 'package.json'))).toString());
-    const packageOpenApiSupportedVersion: string | undefined = packageJsonFile.openApiSupportedVersion?.replace(/\^|~/, '');
-    let openApiVersion = '';
-    try {
-      openApiVersion = (tree.readJson(path.posix.join(targetPath, 'openapitools.json')) as any)?.['generator-cli']?.version;
-    } catch {
-      context.logger.warn('No openapitools.json file found in the project');
-    }
-    if (!!packageOpenApiSupportedVersion && semver.valid(packageOpenApiSupportedVersion) && (!packageOpenApiSupportedVersion || !semver.satisfies(openApiVersion, packageOpenApiSupportedVersion))) {
-      generatorOptions.generatorVersion = packageOpenApiSupportedVersion;
-    }
-    if (options.specConfigPath) {
-      generatorOptions.specConfigPath = options.specConfigPath;
-    }
-    return () => (new OpenApiCliGenerator(options)).getGeneratorRunSchematic(generatorOptions, {rootDirectory: options.directory || undefined});
-  };
+    const runGeneratorRule = () => {
+      return () => (new OpenApiCliGenerator(options)).getGeneratorRunSchematic(
+        (options.generatorKey && JAVA_OPTIONS.every((optionName) => options[optionName] === undefined)) ? {generatorKey: options.generatorKey} : generatorOptions,
```

Comment:
in the case we have `generatorKey` defined and one of the `JAVA_OPTIONS` defined we can raise a warning if one of the options in the `openapitools.json` will not be taken into account (for example: `gitUserId`)